src/main/java/frc/robot/subsystems/chassis/Chassis.java

```diff
@@ -134,7 +150,7 @@ public void setAngleAndVelocity(double vx, double vy, double angle) {
      *               back left, back right
      */
     private void setModuleStates(SwerveModuleState[] states) {
-        SwerveDriveKinematics.desaturateWheelSpeeds(states, SwerveConstants.MAX_SPEED);
+        SwerveDriveKinematics.desaturateWheelSpeeds(states, ChassisConstants.MAX_SPEED);
         for (int i = 0; i < 4; i++) {
             states[i] = SwerveModuleState.optimize(states[i], modules[i].getAngleRotation());
             modules[i].setState(states[i]);
```

Comment:
why not make it streamed?
if there isn't a good reason please do.

Reply:
you need to synchronize the index for states and modules
there might be a way to stream to arrays together but I don't know of it